sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -1105,15 +1123,15 @@ internal async Task<byte[]> GetStateInternal(TimeSpan timeout)
             IList<ServiceBusReceivedMessage> messages = null;
             await _retryPolicy.RunOperation(
                 async (timeout) => messages = await ReceiveDeferredMessagesAsyncInternal(
-                    sequenceNumbers.ToArray(),
+                    sequenceNumbers,
                     timeout).ConfigureAwait(false),
                 ConnectionScope,
                 cancellationToken).ConfigureAwait(false);
             return messages;
         }
 
         internal virtual async Task<IList<ServiceBusReceivedMessage>> ReceiveDeferredMessagesAsyncInternal(
-            long[] sequenceNumbers,
+            IList<long> sequenceNumbers,
```

Comment:
Same assumptions here. I see no reason to copy because the list cannot be reused before the operation is done

Reply:
Same as above. ListEncoding and ArrayEncoding are differnt and Service Bus service expects ArrayEncoding here so we need to revert this.